test/Microsoft.AspNet.Mvc.Core.Test/Formatters/InputFormatterTest.cs

```diff
@@ -0,0 +1,292 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Http.Internal;
+using Microsoft.AspNet.Mvc.ModelBinding;
+using Microsoft.Net.Http.Headers;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc.Formatters
+{
+    public class InputFormatterTest
+    {
+        private class CatchAllFormatter : TestFormatter
```

Comment:
```
for (var i = 0; i < Int32.MaxValue; i++)
{
    Console.WriteLine("Thank you!");
}
```

is deserved for writing these tests in a way that doesn't depend on the `JsonFormatter` implementation.